integration-tests/tests/client/process_blocks.rs

```diff
@@ -3331,6 +3331,106 @@ mod access_key_nonce_range_tests {
+            1,
+            "test1".parse().unwrap(),
+            implicit_account_id.clone(),
+            &signer1,
+            10u128.pow(23),
+            *genesis_block.hash(),
+        );
+        let delete_account_tx = SignedTransaction::delete_account(
+            2,
+            "test1".parse().unwrap(),
+            implicit_account_id.clone(),
+            "test0".parse().unwrap(),
+            &signer1,
+            *genesis_block.hash(),
+        );
+        let send_money_again_tx = SignedTransaction::send_money(
+            3,
+            "test1".parse().unwrap(),
+            implicit_account_id.clone(),
+            &signer1,
+            10u128.pow(23),
+            *genesis_block.hash(),
+        );
+        let send_money_from_implicit_account_tx = SignedTransaction::send_money(
+            1,
+            implicit_account_id.clone(),
+            "test0".parse().unwrap(),
+            &implicit_account_signer,
+            100,
+            *genesis_block.hash(),
+        );
+
+        // Send money to implicit account, invoking its creation.
+        env.clients[0].process_tx(send_money_tx, false, false);
+        for i in 1..4 {
+            env.produce_block(0, i);
+        }
+
+        // Delete implicit account.
+        env.clients[0].process_tx(delete_account_tx, false, false);
+        // Send money to implicit account again, invoking its second creation.
+        env.clients[0].process_tx(send_money_again_tx, false, false);
+        // Send money from implicit account with incorrect nonce.
+        env.clients[0].process_tx(send_money_from_implicit_account_tx, false, false)
+    }
+
+    /// Test that duplicate transactions from implicit accounts are properly rejected.
+    #[cfg(feature = "protocol_feature_access_key_nonce_for_implicit_accounts")]
+    #[test]
+    fn test_transaction_hash_collision_for_implicit_account_fail() {
+        let protocol_version =
+            ProtocolFeature::AccessKeyNonceForImplicitAccounts.protocol_version();
+        let res = get_result_transaction_hash_collision_for_implicit_account(protocol_version);
+        assert!(matches!(
+            res,
+            NetworkClientResponses::InvalidTx(InvalidTxError::InvalidNonce { .. })
+        ));
+    }
+
+    /// Test that duplicate transactions from implicit accounts are not rejected until protocol upgrade.
+    #[test]
+    fn test_transaction_hash_collision_for_implicit_account_ok() {
+        #[cfg(feature = "protocol_feature_access_key_nonce_for_implicit_accounts")]
+        let protocol_version =
+            ProtocolFeature::AccessKeyNonceForImplicitAccounts.protocol_version() - 1;
+        #[cfg(not(feature = "protocol_feature_access_key_nonce_for_implicit_accounts"))]
+        let protocol_version = PROTOCOL_VERSION;
+        let res = get_result_transaction_hash_collision_for_implicit_account(protocol_version);
+        assert!(matches!(res, NetworkClientResponses::ValidTx));
+    }
```

Comment:
Could we also make sure that transactions that are supposed to succeed actually succeed?